packages/block-library/src/playlist/view.js

```diff
@@ -0,0 +1,60 @@
+/**
+ * WordPress dependencies
+ */
+import { store, getContext, getElement } from '@wordpress/interactivity';
+
+const { state } = store(
+	'core/playlist',
+	{
+		state: {
+			get currentTrack() {
+				const { currentId } = getContext();
+				return state.tracks[ currentId ];
+			},
+			get isCurrentTrack() {
+				const { currentId, id } = getContext();
+				return currentId === id;
+			},
+		},
+		actions: {
+			changeTrack() {
+				const context = getContext();
+				context.currentId = context.id;
+			},
+			isPlaying() {
+				const context = getContext();
+				context.isPlaying = true;
+			},
+			isPaused() {
+				const context = getContext();
+				context.isPlaying = false;
+			},
+			nextSong() {
+				const context = getContext();
+				const currentIndex = context.tracks.findIndex(
+					( id ) => id === context.currentId
+				);
+				const nextTrack = context.tracks[ currentIndex + 1 ];
+				if ( nextTrack ) {
+					context.currentId = nextTrack;
+					const { ref } = getElement();
+					// Waits a momet before changing the track, since
+					// immediately changing the track can be jarring.
+					setTimeout( () => {
+						ref.play();
+					}, 1000 );
+				}
+			},
+		},
+		callbacks: {
+			autoPlay() {
+				const context = getContext();
+				const { ref } = getElement();
+				if ( context.currentId && context.isPlaying ) {
+					ref.play();
+				}
+			},
```

Comment:
This callback makes it so that if there's a song playing and another one is clicked, it will also start playing automatically. I’m not sure if it’s something you had or not. I think you did.

Reply:
The only thing is that there used to be a pause when the track was switched, because the experience of immediately starting to play the next song was a little unpleasant.